src/shim/sdk/UserSDK.ts

```diff
@@ -34,30 +34,44 @@ export function rotateMasterKey(passcode: string) {
 }
 
 /**
+ * @deprecated Use deleteDevice with no arguments to get the same behavior.
  * Clears local device keys from the current browser instance. This will require the user to enter their passcode the next time they want to use this browser on this machine.
  * This method should usually be called whenever the current user logs out of your application.
  */
-export function deauthorizeDevice() {
+export const deauthorizeDevice = () => deleteDevice().then((deletedDevice) => ({transformKeyDeleted: deletedDevice > 0}));
+
+/**
+ * Deletes a device. If deleting the current device, the user will have to enter their passcode the next time they want to use this browser on this machine.
+ * This method should usually be called whenever the current user logs out of your application or you're aware a device of their's shouldn't have access.
+ * @param {number | undefined} deviceId The device id to delete. If undefined, the current device will be deleted and local storage will be cleared.
+ */
+// TODO: detect if deviceId is the same as the currently operating device and set deletingCurrentDevice to true
+export const deleteDevice = (deviceId?: number) => {
     checkSDKInitialized();
-    const payload: MT.DeauthorizeDevice = {
-        type: "DEAUTHORIZE_DEVICE",
-        message: null,
+    const deletingCurrentDevice = deviceId === undefined;
+    const payload: MT.DeleteDevice = {
+        type: "DELETE_DEVICE",
+        message: deletingCurrentDevice ? undefined : deviceId,
     };
-    //Clear the local symmetric key from local storage, then send a request to clear the frames local storage. Once that's complete clear the SDK init flag
-    //so that the user has to rerun init before the SDK methods will work again.
-    clearParentWindowSymmetricKey();
-    return FrameMediator.sendMessage<MT.DeauthorizeDeviceResponse>(payload)
+    // If current device, clear the local symmetric key from local storage, then send a request to clear the frames local storage.
+    // Once that's complete clear the SDK init flag so that the user has to rerun init before the SDK methods will work again.
+    if (deletingCurrentDevice) {
+        clearParentWindowSymmetricKey();
+    }
+    return FrameMediator.sendMessage<MT.DeleteDeviceResponse>(payload)
         .map(({message}) => {
-            clearSDKInitialized();
-            return {transformKeyDeleted: message};
+            if (deletingCurrentDevice) {
+                clearSDKInitialized();
+            }
```

Comment:
Could this just be in the above after clearing the window key?

Reply:
No, because we need the SDK initialized still to make the delete request that we're sending to the frame here.